### PR TITLE
inter: 3.5 -> 3.7

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Hot on the tail of 3.6, 3.7 is available!
Get yours now before supplies run out! :P

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/data/fonts/inter/default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 { lib, fetchzip }:
 
 let
-  version = "3.5";
+  version = "3.7";
 in fetchzip {
```
Comment (mmahut) [inline pkgs/data/fonts/inter/default.nix]: Wouldn't be it better to use fetchFromGitHub?

File: pkgs/data/fonts/inter/default.nix [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
   name = "inter-${version}";
```
Comment (mmahut) [inline pkgs/data/fonts/inter/default.nix]: Can we move it to ``pname``?

Comment (dtzWill) [reply pkgs/data/fonts/inter/default.nix]: I didn't think fetchzip recognized `pname`, I'll check now and yes if so then that sounds good.

As for fetchFromGitHub--that will work if the sources have the needed files but sometimes (esp re:fonts) there are downloads attached to a release that are basically "built" (although AFAIK they're whatever they want to put, but you know) and these can't be fetched with fetchFromGitHub.  Well at least AFAIK/in the past :).

Comment (dtzWill) [reply pkgs/data/fonts/inter/default.nix]: Update:

* fetchzip does not seem to support `pname` (nor does it accept `version` attribute, which also is a bit of a long-standing "quirk" motivating the `let version = "x.y.z"; in fetchzip ...` pattern.
* upstream's repo is indeed of the sort I describe, so no fetchFromGitHub here without also deciding to build the font ourselves which I'd prefer not to do in general especially for carefully managed build pipelines like they appear to use -- hard to replicate in a nix-y way and quite likely to end up with slightly different and probably inferior results.  However if someone wants to make this happen I'm not outright against it and would be happy to move to that after review.

File: pkgs/data/fonts/inter/default.nix (hunk 1 above)
Comment (dtzWill) [reply pkgs/data/fonts/inter/default.nix]: Answered in other comment, sorry about that.  In short: can't use fetchFromGitHub to grab release assets unfortunately, but if we could yes that's preferred :).

**Comment by dtzWill** [comment]
(merging, but concerns and such still encouraged -- especially if there's a disagreement re:resolution of comments above!)

:heart:

**Pull request merged**